Src/IronPython/Runtime/Operations/PythonOps.cs

```diff
@@ -1381,17 +1303,7 @@ public static partial class PythonOps {
             if (dict.TryGetValue("__metaclass__", out ret) && ret != null) return ret;
 
             // Otherwise, if there is at least one base class, its metaclass is used
-            for (int i = 0; i < bases.__len__(); i++) {
-                if (!(bases[i] is OldClass)) return DynamicHelpers.GetPythonType(bases[i]);
-            }
-
-            // Otherwise, if there's a global variable named __metaclass__, it is used.
-            if (context.TryGetGlobalVariable("__metaclass__", out ret) && ret != null) {
-                return ret;
-            }
-
-            //Otherwise, the classic metaclass (types.ClassType) is used.
-            return TypeCache.OldInstance;
+            return DynamicHelpers.GetPythonType(bases[0]);            
```

Comment:
Yes, new style classes (the only kind in 3.0) always inherit from object. I need to add the check in MakeClass to see if object was already passed as a base.